qt/widgets/common/src/AlgorithmProgress/AlgorithmProgressWidget.cpp

```diff
@@ -0,0 +1,63 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "MantidQtWidgets/Common/AlgorithmProgress/AlgorithmProgressWidget.h"
+#include "MantidQtWidgets/Common/AlgorithmProgress/AlgorithmProgressDialogWidget.h"
+#include "MantidQtWidgets/Common/AlgorithmProgress/AlgorithmProgressPresenter.h"
+
+#include <QProgressBar>
+
+namespace MantidQt {
+namespace MantidWidgets {
+
+AlgorithmProgressWidget::AlgorithmProgressWidget(QWidget *parent)
+    : QWidget(parent),
+      m_progressBar{new QProgressBar(this)}, m_layout{new QHBoxLayout(this)},
+      m_detailsButton{new QPushButton("Details")},
+      m_presenter{std::make_unique<AlgorithmProgressPresenter>(parent, this)}
+
+{
+  setAttribute(Qt::WA_DeleteOnClose);
```

Comment:
Similar comment to the details one. Should this not be set by the code that creates the widget?

Reply:
I believe that it should stay here as otherwise in Python it would have a very odd position. Specifically in widget.py.

![image](https://user-images.githubusercontent.com/24723609/54363294-97db9400-4662-11e9-84cc-b956ae16f5f2.png)

Reply:
The reason behind it being needed to be done in python is that the widget itself owns the presenter so the presenter would not be making the widget.